src/main/java/net/sf/marineapi/nmea/io/DefaultDataReader.java

```diff
@@ -52,6 +52,10 @@ public DefaultDataReader(InputStream source, SentenceReader parent) {
 	 */
 	@Override
 	public String read() throws Exception {
-		return input.readLine();
+		final String result = input.readLine();
+		if (result == null) {
+		    setRunning(false);

```

Comment:
We can't stop here on first `null`, because there might be still more data coming in.